vigranumpy/src/core/segmentation.cxx

```diff
@@ -1147,6 +1148,65 @@ pythonApplyMapping(NumpyArray<NDIM, Singleband<SrcVoxelType> > src,
 //VIGRA_PYTHON_MULTITYPE_FUNCTOR_NDIM(pyApplyMapping, pythonApplyMapping)
 
 
+/** \brief Find unique values in the given array.
+*/
+template <class VoxelType, unsigned int NDIM>
+NumpyAnyArray
+pythonUnique(NumpyArray<NDIM, Singleband<VoxelType> > src)
+{
+    boost::unordered_set<VoxelType> labelset;
+    auto f = [&labelset](VoxelType px) { labelset.insert(px); };
+    inspectMultiArray(src, f);
+
+    NumpyArray<1, VoxelType> result;
+    result.reshape( Shape1(labelset.size()) );
+    std::copy( labelset.begin(), labelset.end(), result.begin() );
+    return result;
+}
+
+VIGRA_PYTHON_MULTITYPE_FUNCTOR_NDIM(pyUnique, pythonUnique)
+
+
+/** \brief Relabel an array such that all labels are consecutive
+ * (i.e. there are no gaps in the label values used by the array)
+ * See python docstring below for details.
+*/
+template <unsigned int NDIM, class SrcVoxelType, class DestVoxelType>
+NumpyAnyArray
```

Comment:
I think the function should also return the new max label and the mapping (converted to a Python dict).